### PR TITLE
the "remove all sketches" and "convert annotations to memory layer" b…

### PR DESCRIPTION
**Pull request by Fighterisr**
fixed issues #12  and #15

### CODE REVIEW

File: redLayerModule.py [hunk 1]
```diff
@@ -368,6 +368,7 @@ def removeSketchesAction(self):
                     self.iface.mapCanvas().scene().removeItem( sketch[3] )
                 except:
                     pass
+        self.removeAllAnnotations()
```
Comment (enricofer) [inline redLayerModule.py:371]: Why are you removing all annotations when using eraser tool? It should erase only the pointed annotation, linked to the erasing segment and pointed by sketch[3] see line 368

File: redLayerModule.py [hunk 2]
```diff
@@ -368,6 +368,7 @@ def removeSketchesAction(self):
                     self.iface.mapCanvas().scene().removeItem( sketch[3] )
```
Comment (enricofer) [inline redLayerModule.py:368]: Anyway this line does not remove the pointed annotation. Could you investigate why?

**Review by enricofer — CHANGES_REQUESTED** [review]
Thanks for submission.
I gave a look to PR and I found issues related to erasing single annotation.
I fear that QgsAnnotation handling API break causes regressions even on saving and loading existing sketches with annotations.
Could you give a look and make some tests?

**Comment by Fighterisr** [comment]
sadly i am very busy right now
i thought that the "erase all button" is supposed to delete everything, including annotations, so i included the call in removeSketchesAction()

the point of this commit was to fix the removeAllAnnotations() which did try to remove annotations from the scene but didn't work at all. mapCanvas().annotationItems() seems like the right way.

maybe in the future i'll try again

**Comment by enricofer** [comment]
I published a new plugin release 2.0.4 with all fixes for QgsAnnotation API break. Thanks for your submissions @Fighterisr @haubourg @SebastienPeillet :https://plugins.qgis.org/plugins/redLayer/

**Comment by haubourg** [comment]
👍 Thanks @enricofer !

**Pull request closed without merging**